api/lib/source/polkadotV0-source.js

```diff
@@ -114,23 +74,27 @@ class polkadotAPI {
   async getBlockByHeightV2(blockHeight) {
     const api = await this.getAPI()
 
+    let blockHash
+    if (blockHeight) {
+      blockHash = await api.rpc.chain.getBlockHash(blockHeight)
+    } else {
+      blockHash = await api.rpc.chain.getFinalizedHead()
+    }
     // heavy nesting to provide optimal parallelization here
     const [
-      [{ author }, { block }, blockEvents, blockHash],
+      { author, number },
+      { block },
+      blockEvents,
       sessionIndex
     ] = await Promise.all([
-      api.rpc.chain.getBlockHash(blockHeight).then(async (blockHash) => {
-        const [{ author }, { block }, blockEvents] = await Promise.all([
-          api.derive.chain.getHeader(blockHash),
-          api.rpc.chain.getBlock(blockHash),
-          api.query.system.events.at(blockHash)
-        ])
-
-        return [{ author }, { block }, blockEvents, blockHash]
-      }),
+      api.derive.chain.getHeader(blockHash),
+      api.rpc.chain.getBlock(blockHash),
+      api.query.system.events.at(blockHash),
```

Comment:
What are all these changes? Seems quite beyond the initial scope of the PR, namely, just adding changes to description.

This makes the PR difficult to read